src/index.js

```diff
@@ -25,6 +25,8 @@ export default function Knex(config) {
   let Dialect;
   if (arguments.length === 0 || (!config.client && !config.dialect)) {
     Dialect = makeClient(Client)
+  } else if (typeof config.client === 'function' && config.client.prototype instanceof Client) {
```

Comment:
This can be simplified to:

```js
} else if (config.client instanceof Client)
```

`instanceof` will walk the prototype list. It will return `false` if `config.client` is not a function.